### DISCUSSION TITLE
How to deploy a pathology pytorch model on MONAI label apps?

### DISCUSSION THREAD

**Discussion by OmarAshkar** [Project-MONAI/MONAILabel#1165] [Q&A]
Hello,

I am wondering how may I customize the pathology sample apps with my own model, so I can interact with the model more and fix it if needed.

My best guess is the start will be where the pretrained model is loaded in monai label.

Thanks for your great efforts!
Omar

**Comment by tangy5** [comment]
Hi @OmarAshkar , yes, you could start creating a new model or customized the pathology app by getting familiar with  the current template app. 
Such as the codes in the `lib` folder, "segmentation_nuclei" as an example, it's needed to define basic things (networks, parameter, etc) in the `config`, then add inferer and trainer for `infers` and `trainers` folder for specific transforms. 
https://github.com/Project-MONAI/MONAILabel/tree/main/sample-apps/pathology/lib

Pretrained model is set here: 
https://github.com/Project-MONAI/MONAILabel/blob/6007e0ca6e3cbee17f14e4d99cc51c44a4d8949c/sample-apps/pathology/lib/configs/segmentation_nuclei.py#L66

Finally, adjust the integration in the `main.py` if the loading process or initialization steps need any change:
https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/pathology/main.py

Thanks.

**Comment by SenhorUnk** [comment]
Would be super helpful to have a step-by-step-video instruction on how to implement own models. Still not getting it 100%. My focus is currently on the endoscopy sample app since my use case is similar. I am trying to follow the steps in the documentation (https://github.com/Project-MONAI/MONAILabel/tree/main/sample-apps/endoscopy#how-to-add-new-model) but I don't really understand the strucuture of the MONAI Label App and how the internal workflow of inference looks like. 

Additionally, the referenced endoscopy/lib/trainers/tooltracking.py does not hold much information that could be used as a template.

**Reply by tangy5** [reply]
Thanks, we are making and putting up more tutorials/doc.
Videos are also good idea. We are working on getting started with MONAI Label easier. 
Thanks for feedback.

**Reply by SenhorUnk** [reply]
Thanks :)
